lib/skate/detours/missed_stops.ex

```diff
@@ -92,99 +93,63 @@ defmodule Skate.Detours.MissedStops do
           shape :: [Util.Location.From.t()],
           stops :: [Util.Location.From.t()]
         ) :: [Skate.Detours.ShapeSegment.t()]
-  defp segment_shape_by_stops([] = _shape, [] = _stops),
+  def segment_shape_by_stops([] = _shape, [] = _stops),
```

Comment:
thought(non-blocking/address in another PR): IIRC we originally considered this a implementation detail that shouldn't be tested, now that it's being tested, should this become a generic "segment shape by projected points" that's generic across `Util.Location.From` elements so we retain the same functionality?

I _think_ we could then use this for both route segments and missed stops?

Reply:
Like, I think the important thing about this function is that it's "generic" across inputs as long as they implement `Util.Location.From`, and that this function allows extra data to be stuffed in with the algorithm's return value.

Reply:
Ooooh I like that!

Definitely fodder for a follow-up refactor PR. Will queue that up on my own personal task-list.